### PR TITLE
Debugging improvements for #974

### PR DESCRIPTION
**Pull request by jansel**
Partial fix to get to the next error in #974, plus some extra debugging I used to diagnose the issue.

### CODE REVIEW

File: torchdynamo/debug_utils.py [hunk 1]
```diff
@@ -194,6 +194,7 @@ def nvfuser_fails(fx_g, args, check_str=None):
 
 
 def dump_to_minify(gm, args, compiler_name: str):
+    favored_device = 1 if torch.cuda.device_count() >= 2 else 0
```
Comment (desertfire) [inline torchdynamo/debug_utils.py:197]: Yes, I just figured this out yesterday.

**Comment by desertfire** [comment]
A unit test needs to be fixed.

**Comment by jansel** [comment]
> A unit test needs to be fixed.

This doesn't actually fix this issue.  It just gets to the next error (with bettor debug printouts).

File: torchinductor/compile_fx.py [hunk 2]
```diff
@@ -130,6 +131,10 @@ def compile_fx_inner(
         elif graph.mutated_inputs and set(graph.device_types) == {"cuda"}:
             log.warning("skipping cudagraphs due to input mutation")
 
+    if torchdynamo.config.repro_level > 1:
```
Comment (Chillee) [inline torchinductor/compile_fx.py:134]: Can we put this in the `wrap_debug` utility?

**Review by Chillee — APPROVED** [review]

**Pull request merged**